### PR TITLE
issue 11668 : re-initialize calculated waypoint correctly

### PR DESCRIPTION
**Pull request by pet0etie**
## Description

corrections in CoordinatesCalculateDialog.java
on second visit we enter with empty gp ... so the-not-formula-part of the coords are initialised with those of your current location
re-initialisation is now done correctly and depending on the saved coordinate-format

## Related issues

issue #11668

## Additional context

checkstyle-test : no remarks => ok
translations : none => ok

### CODE REVIEW

**Comment by Lineflyer** [comment]
Thanks @pet0etie for your contribution.
Without having an idea of the involved code part, I wonder why the fix is so "extensive" while the bug showed up suddenly after recent changes. But maybe I am wrong and it never worked?

Remark regarding syntax:
You might want to use the keyword "Fix #11668" (ideally in your commit message) to allow auto-closing of the issue on merge.
Also when referring to an issue in your text just prepend a "#" to generate an auto-link.

**Comment by ztNFny** [comment]
I have to admit I didn't look into the root cause of the problem, but the code in this PR seems to be very "scripty". Like using "" to initialize a string instead of making sure you get a string and the explicit indexes should be in a loop instead.
Applying different solutions in different cases seems to also be prone to cause future issues. I think there must be a more elegant solution to this issue.

File: main/src/cgeo/geocaching/ui/dialog/CoordinatesCalculateDialog.java [hunk 1]
```diff
@@ -329,6 +329,9 @@ public void onCreate(final Bundle savedInstanceState) {
         gp = getArguments().getParcelable(GEOPOINT_ARG);
         if (gp == null) {
             gp = Sensors.getInstance().currentGeo().getCoords();
+            if (savedState != null) {
+                gp = new Geopoint(savedState.plainLat, savedState.plainLon);
+            }
```
Comment (bekuno) [inline main/src/cgeo/geocaching/ui/dialog/CoordinatesCalculateDialog.java:334]: If savedState != null then you initialize gp twice.
Better e.g.:
~~~
 if (savedState != null) {
    gp = new Geopoint(savedState.plainLat, savedState.plainLon);
} else {
    gp = Sensors.getInstance().currentGeo().getCoords();
}
~~~

**Comment by pet0etie** [comment]
after analyzing the code again everything starts with a null-gp when getting back into the CoordinateCalculateDialog again
my first thought was rather correct : i could "retrieve the original coordinates" from the savedState
but i used the "button-values" so it was indeed a hard time to get good coordinates from there
the current solution uses the savedState too, but uses the "plainLat" and "plainLon" 
so i ended up in only 3 lines of code ...

File: main/src/cgeo/geocaching/ui/dialog/CoordinatesCalculateDialog.java (hunk 1 above)
Comment (pet0etie) [reply main/src/cgeo/geocaching/ui/dialog/CoordinatesCalculateDialog.java]: suggestion was changed in last commit/push

**Pull request merged**